config/dataset/coral.yaml

```diff
@@ -0,0 +1,5 @@
+name: CoRal_raw
+id: CoRal_raw
+input_path: data/raw
+metadata_path: data/raw/metadata.csv
+output_path: data/processed
```

Comment:
I'm not sure if this configuration file makes sense here, as the dataset configs are currently meant to be configurations for datasets that already exists on the Hugging Face Hub.

This PR is about building a new dataset, which I think belongs as a script instead of a module, since we only have to do it once (in theory), and future users of this repo won't really need functions related to dataset building, as the datasets should already exist at that point (and that they don't have access to the raw files anyway).